### ISSUE TITLE
failed to run weston-terminal in orbital

### ISSUE THREAD

**Issue by xiangzhai** [giucam/orbital#16]
weston-terminal: clients/window.c:4538: window_create_internal: Assertion `custom || display->xdg_shell || display->ivi_application' failed.

But in weston, it is ok!

**Comment by giucam** [comment]
Known issue, orbital doesn't implement xdg_shell yet.

**Comment by xiangzhai** [comment]
Hi @giucam 

I just simply add xdg-shell skeleton https://github.com/xiangzhai/orbital/commit/1d5d6b05249a2edcabb13f316982174e85fec2df

It is just be familiar with wayland && weston relative thing ;)

I will follow the weston`s source code http://cgit.freedesktop.org/wayland/weston/tree/desktop-shell/shell.c#n3951

Regards,
Leslie Zhai

**Comment by xiangzhai** [comment]
Hi @giucam 

How to implement xdg-shell correctly?

There are already XdgShell https://github.com/xiangzhai/orbital/blob/xdg-shell/src/compositor/shell.cpp#L69 and XdgShellSurface && XdgShellPopup https://github.com/xiangzhai/orbital/tree/xdg-shell/src/compositor/xdg-shell

But weston-terminal is still unvisible! Please give me some advice, thanks a lot!

Regards,
Leslie Zhai

**Comment by xiangzhai** [comment]
Oops~~~ xdg-shell resize segfault https://github.com/xiangzhai/orbital/issues/1

**Comment by gld1982ltd** [comment]
Is this going to get implemented?

**Comment by giucam** [comment]
Yes, but it is not high priority for me at the moment. I'll gladly review a PR though.

**Comment by Icenowy** [comment]
@xiangzhai is working on a xdg_shell implementation on Orbital. https://github.com/xiangzhai/orbital but it's still buggy, only able to run weston demo clients including weston-terminal.

**Comment by xiangzhai** [comment]
@Icenowy I am developing some close source stuff now ;-)

**Comment by xiangzhai** [comment]
Hi @Icenowy I have finished the ***CLOSE*** source stuff, then take a break to fix the xdg-shell issue for running gtk3-demo ;-)

![orbital-xdg-shell-gtk3-demo](https://cloud.githubusercontent.com/assets/851100/7040754/fff21048-de04-11e4-99ab-bdc5f611c2c5.png)

**Comment by xiangzhai** [comment]
Hi @wengxt what about Fcitx for Wayland branch? I saw a video in 2014 about your fcitx working under wayland ;-) can I join the wayland branch`s stuff? I wanna switch to wayland now, X11 is toooooooo ***OLD***!

**Comment by gld1982ltd** [comment]
@xiangzhai AWESOME! Looking forward to that.

**Comment by xiangzhai** [comment]
@gld1982ltd happy hacking ;P and I am learning how to develop weston plugin https://github.com/xiangzhai/weston/blob/sheep/README and tried to migrate RandR http://lists.freedesktop.org/archives/wayland-devel/2015-April/021199.html and implement mirroring mode for weston http://lists.freedesktop.org/archives/wayland-devel/2015-April/021245.html

**Comment by giucam** [comment]
Hi @xiangzhai. Do you plan to make a pull request of your xdg-shell implementation?
